templates/terraform/constants/firebase_project.go.erb

```diff
@@ -0,0 +1,57 @@
+func resourceFirebaseProjectFinalizeDefaultLocation(d *schema.ResourceData, meta interface{}) error {
+	prevLocationId, locationId := d.GetChange("resources.0.location_id")
+	if locationId != nil && len(locationId.(string)) > 0 {
+		config := meta.(*Config)
+
+		if prevLocationId != nil && len(prevLocationId.(string)) > 0 {
+			return fmt.Errorf(
+				"Unable to change default location of a Firebase project (%s) after it has been set (%s)",
+				locationId,
+				prevLocationId)
+		}
+
+		obj := map[string]interface{}{
+			"locationId": locationId,
+		}
+
+		url, err := replaceVars(d, config, "{{FirebaseBasePath}}projects/{{project}}/defaultLocation:finalize")
+		if err != nil {
+			return err
+		}
+		project, err := getProject(d, config)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] Finalizing project location (%s): %#v", project, obj)
+		res, err := sendRequestWithTimeout(config, "POST", project, url, obj, d.Timeout(schema.TimeoutUpdate))
+		if err != nil {
+			return fmt.Errorf("Error finalizing default project default location: %s", err)
+		}
+		err = firebaseOperationWaitTime(
+			config, res, project, "Setting Project Location",
+			int(d.Timeout(schema.TimeoutUpdate).Minutes()))
+
+		if err != nil {
+			return fmt.Errorf("Error waiting to finalize default project default location: %s", err)
+		}
+		log.Printf("[DEBUG] Finished setting project default location %q: %#v", d.Id(), res)
+	}
+	return nil
+}
+
+func flattenFirebaseProjectResources(v interface{}) interface{} {
+	if v == nil {
+		return nil
+	}
+	original := v.(map[string]interface{})
+	if len(original) == 0 {
+		return nil
+	}
+	transformed := make(map[string]interface{})
+	transformed["hosting_site"] = original["hostingSite"]
```

Comment:
The generated flatteners should already handle the renaming of the fields. I don't think you should need a custom flattener at all.

Reply:
Unfortunately as stated above, if I mark the property as excluded in terraform.yaml, I had to define the flatteners in custom code.  Those are actually the generated flatteners but just moved into custom code after I marked the resources property as excluded.